File: java/2025/PassGen.java
import java.util.Scanner ;
import java.util.Random ;

public class PassGen {
    public static void main(String[] args) {
        String chars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!#$()" ;
        Scanner scan = new Scanner(System.in) ;
        Random rand = new Random() ;

        System.out.print("Length of password: ") ;
        int passLen = scan.nextInt() ;
        System.out.print("How many passwords: ") ;
        int passNum = scan.nextInt() ;

        for(int k = 0 ; k < passNum ; k++) {
            String passwd = "" ;
            for (int i = 0; i < passLen; i++) {
                int j = rand.nextInt(0, chars.length());
                passwd += chars.charAt(j);
            }

            System.out.println("Generated Password: " + passwd);
        }
    }
}

File: c/pass_gen.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(void) 
{
    char* chars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!#$()" ;
    
    int chars_len = 0 ;
    while(chars[chars_len] != '\0')
    {
        chars_len++ ;
    }

    int passLen = 0 , passNum = 0 ;
    printf("Length of password: ") ;
    scanf("%d" , &passLen) ;
    printf("How many passwords: ") ;
    scanf("%d" , &passNum) ;

    char passwd[passLen] ;
    for(int i = 0 ; i < passNum ; i++)
    {
        // clearing the passwd[] before each iteration
        memset(passwd , 0 , passLen * (sizeof passwd[0])) ;
        for(int j = 0 ; j < passLen ; j++)
        {
            int k = rand() % chars_len ;
            passwd[j] = chars[k] ;
        }

        printf("Generated Password: %s\n" , &passwd) ;
    }

    return 0 ;
}
